app/src/main/java/com/coopra/nebulus/view_models/AlternativeViewModel.java

```diff
@@ -0,0 +1,51 @@
+package com.coopra.nebulus.view_models;
+
+import android.app.Application;
+import android.os.AsyncTask;
+
+import com.coopra.data.Track;
+import com.coopra.nebulus.TrackRepository;
+
+import java.util.List;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.AndroidViewModel;
+import androidx.lifecycle.MutableLiveData;
+
+public class AlternativeViewModel extends AndroidViewModel {
+    private MutableLiveData<List<Track>> mTracks;
+
+    public AlternativeViewModel(@NonNull Application application) {
+        super(application);
+    }
+
+    public MutableLiveData<List<Track>> getTracks() {
+        if (mTracks == null) {
+            mTracks = new MutableLiveData<>();
+        }
+
+        return mTracks;
+    }
+
+    public void retrieveTracks() {
+        new GetTracksTask(getTracks()).execute();
+    }
+
+    private static class GetTracksTask extends AsyncTask<Void, Void, List<Track>> {
+        MutableLiveData<List<Track>> mTracksLiveData;
```

Comment:
Since you're passing `tracksLiveData` into the constructor and setting in to `mTracksLiveData` and never updating it you should consider making `mTracksLiveData` `final` so that it can't be modified.

```suggestion
        private final MutableLiveData<List<Track>> mTracksLiveData;
```